src/modules/site-v2/templates/data/submit-trait-form.html

```diff
@@ -0,0 +1,185 @@
+        </div>
+        <div class="row my-5">
+            <div class="col-md-4">{{ render_field(form.trait_name_display_3, has_popover=true, content='This field can be used to provide additional information that\'s important for the user to know about the trait.') }}</div>
+            <div class="col-md-4"> {{ render_field(form.description_short ,has_popover=true, content='Please describe the trait in a few words, providing the information that\'s most important for other users who may want to perform an analysis using this trait.') }}</div>
+            <div class="col-md-4">{{ render_field(form.description_long, has_popover=true, content='You may provide a longer description of the trait to provide additional useful information for other users who may want to perform an analysis using this trait.') }}</div>
+        </div>
+        <div class="row my-5">
+            <div class="col-md-4">{{ render_field(form.units) }}</div>
+            <div class="col-md-4">{{ render_field(form.protocols) }}</div>
+            <div class="col-md-4">{{ render_field(form.publication) }}</div>
+        </div>
+        <div class="row my-5">
+            <div class="col-md-4 text-break">{{ render_field(form.tags, has_popover=true, content='Please select the best categories to describe your trait. Please note: at least one category is required.') }}</div>
+        </div>
+        <div class="row my-5">
+        </div>
+        <button type="submit" id="submit-trait" value="Submit Form" class="btn btn-secondary text-light" aria-label="submit">Add Trait</button>
+    </form>
+</div> <!-- /Container -->
+
+{% endblock %}
+
+{% block script %}
+
+<script>
+    {% include '_scripts/utils.js' %} {#/* defines: flash_message */#}
+
+    $(document).ready(function() {
+        
+        // Add 'required' class to labels
+        $('.label-group').each(function() {
+            const requiredSelector = $(this).next().attr('required') && $(this).find('label')
+
+            if (requiredSelector) {
+                requiredSelector.addClass('required')
+            }
+            if ($(this).children('label').text().includes('file')) {
+                $(this).children('label').addClass('required')
+            }
+        })
+
+        $.ajaxSetup({
+            beforeSend: function(xhr, settings) {
+                if (!/^(GET|HEAD|OPTIONS|TRACE)$/i.test(settings.type) && !this.crossDomain) {
+                xhr.setRequestHeader("X-CSRFToken", "{{ form.csrf_token._value() }}")
+                }
+            }
+        });
+
+        $('#speciesSelect, #file').on('change', function() {
+            $('#file-upload').prop('disabled', !$('#speciesSelect').val() || !$('#file').val())
+        })
+
+        $('#file').on('change', function() {
+            $('#file-content').empty()
+            $('#file-content').parent().addClass('d-none')
+        })
+
+        // Upload the file and display the content
+        $('#file-upload').on('click', function(e) {
```

Comment:
one front-end architecture idea (loosely similar to the MVC comment earlier), see if we can decouple the event handling from the action itself; especially when the action is more than a few lines. Consider the following: 

```js

// declare and implement the action as a function 
const handleFileUpload = async (...) => {};

// then use it in the event handler
$('#file-upload').on('click',  [async] (event) => {
  event.preventDefault();
  [await] handleFileUpload(...);
});
```
operators in square brackets are optional, use if needed.

Reply:
Pushed in the commit 640f8f002e1e2a30d087d72f8b3e9f55d1c1d050